### PR TITLE
Added properties - Resolving Issues

### PR DESCRIPTION
**Pull request by haripriyad252**

### CODE REVIEW

**Review by hvarg — COMMENTED** [review]
Maybe `hasName` is not necessary.

**Comment by dgarijo** [comment]
@hvarg if it doesn't hurt, let's keep it for now.

**Comment by dgarijo** [comment]
@haripriyad252 the documentation for the ontology changes is missing.

**Comment by haripriyad252** [comment]
Yes if the new properties are fine then I'll go ahead and include the documentation as well.

File: disk/development/ontology.ttl [hunk 1]
```diff
@@ -131,6 +131,12 @@ xsd:date rdf:type rdfs:Datatype .
                        rdfs:domain :MetaWorkflow ,
                                    :WorkflowBinding .
 
+###  http://disk-project.org/ontology/disk#hasQuestionVariable
+:hasQuestionVariable rdf:type owl:ObjectProperty ;
+               rdfs:domain :Question ;
+               rdfs:range :QuestionVariable ;
+               rdfs:comment "property that links a question to a question variable to store bindings."@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: This definition is not very accurate.
This is a property that links a hypothesis question to a variable that appears in that question. The variable will be a value that can be specified by the line of inquiry or the user.

File: disk/development/ontology.ttl [hunk 2]
```diff
@@ -209,6 +215,24 @@ xsd:date rdf:type rdfs:Datatype .
...
+###  http://disk-project.org/ontology/disk#hasAuthor
+:hasAuthor rdf:type owl:ObjectProperty ;
+             rdfs:domain [ rdf:type owl:Class ;
+                                      owl:unionOf ( :LineOfInquiry
+                                                    :TriggeredLineOfInquiry
+                                                    :Hypothesis
+                                                  )
+                                    ] ;
+             rdfs:range :Person ;
+             rdfs:comment "Property that links a Line of Inquiry, Triggered Line of Inquiry or Hypothesis to its author"@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: property to describe who is the author of a line of inquiry, a triggered line of inquiry or a hypothesis

File: disk/development/ontology.ttl [hunk 3]
```diff
@@ -302,6 +326,53 @@ xsd:date rdf:type rdfs:Datatype .
...
+
+###  http://disk-project.org/ontology/disk#hasUsageNotes
+:hasUsageNotes rdf:type owl:DatatypeProperty ;
+             rdfs:domain [ rdf:type owl:Class ;
+                                      owl:unionOf ( :LineOfInquiry
+                                                    :Hypothesis
+                                                  )
+                                    ] ;
+             rdfs:range xsd:string ;
+             rdfs:comment "Property that add notes to Line of Inquiry and Hypothesis"@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: It doesn't add notes. It's a property to indicate what are the usage notes to take into consideration when using a line of inquiry or a hypothesis

File: disk/development/ontology.ttl [hunk 4]
```diff
@@ -302,6 +326,53 @@ xsd:date rdf:type rdfs:Datatype .
...
+             rdfs:label "has usage notes"@en .
+
+###  http://disk-project.org/ontology/disk#hasQuestionTemplate
+:hasQuestionTemplate rdf:type owl:DatatypeProperty ;
+               rdfs:domain :Question ;
+               rdfs:range xsd:string ;
+               rdfs:comment "property to store human readable template of hypothesis question"@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: This is not grammatical. And is it human readable?
Property designed to store a readable template associated with a hypothesis question

File: disk/development/ontology.ttl [hunk 5]
```diff
@@ -302,6 +326,53 @@ xsd:date rdf:type rdfs:Datatype .
...
+               rdfs:label "has question template"@en .
+
+
+###  http://disk-project.org/ontology/disk#hasVariableName
+:hasVariableName rdf:type owl:DatatypeProperty ;
+               rdfs:domain :QuestionVariable ;
+               rdfs:range xsd:string ;
+               rdfs:comment "property to store a variable name"@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: Property that indicates what is the name of a given variable

File: disk/development/ontology.ttl [hunk 6]
```diff
@@ -302,6 +326,53 @@ xsd:date rdf:type rdfs:Datatype .
...
+               rdfs:label "has variable name"@en .
+
+
+###  http://disk-project.org/ontology/disk#hasConstraints
+:hasConstraints rdf:type owl:DatatypeProperty ;
+               rdfs:domain :QuestionVariable ;
+               rdfs:range xsd:string ;
+               rdfs:comment "property to store a query that will return possible values"@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: Property designed to indicate the constraints of a question variables. For example ....

File: disk/development/ontology.ttl [hunk 7]
```diff
@@ -335,12 +406,16 @@ xsd:date rdf:type rdfs:Datatype .
                   rdfs:label "Confidence report"@en .
 
 
+###  http://disk-project.org/ontology/disk#Person
+:Person rdf:type owl:Class ;
+      rdfs:comment "Contains details about a person"@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: No! identifies persons

File: disk/development/ontology.ttl [hunk 8]
```diff
@@ -388,6 +463,15 @@ xsd:date rdf:type rdfs:Datatype .
                  rdfs:comment "Contains a workflow and a variable binding"@en ;
                  rdfs:label "Workflow binding"@en .
 
+###  http://disk-project.org/ontology/disk#Question
+:Question rdf:type owl:Class ;
+                 rdfs:comment "Contains a hypothesis question"@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: You are not defining question here. You are using the same term in the definition

File: disk/development/ontology.ttl [hunk 9]
```diff
@@ -388,6 +463,15 @@ xsd:date rdf:type rdfs:Datatype .
...
+                 rdfs:label "Question"@en .
+
+###  http://disk-project.org/ontology/disk#QuestionVariable
+:QuestionVariable rdf:type owl:Class ;
+                 rdfs:comment "Contains a hypothesis question variable"@en ;
```
Comment (dgarijo) [inline disk/development/ontology.ttl]: same as above. What is a QuestionVariable?

**Review by dgarijo — CHANGES_REQUESTED** [review]

**Comment by dgarijo** [comment]
I reviewed the definitions. Please not just copy paste the suggestions. Read them slowly because there are some that require some extra work. 
Once done, please also add the documentation

**Pull request merged**